homeassistant/components/lifx/binary_sensor.py

```diff
@@ -50,15 +51,8 @@ def __init__(
         super().__init__(coordinator)
         self.entity_description = description
         self._attr_unique_id = f"{coordinator.serial_number}_{description.key}"
-        self._async_update_attrs()
 
-    @callback
-    def _handle_coordinator_update(self) -> None:
-        """Handle updated data from the coordinator."""
-        self._async_update_attrs()
-        super()._handle_coordinator_update()
-
-    @callback
-    def _async_update_attrs(self) -> None:
-        """Handle coordinator updates."""
-        self._attr_is_on = self.coordinator.async_get_hev_cycle_state()
+    @property
+    def is_on(self) -> bool | None:
+        """Returns true if a HEV cycle is currently running."""
+        return self.coordinator.async_get_hev_cycle_state()
```

Comment:
Unless I'm missing something it doesn't look like the platform changes are material to this change as its effectively going to do the same thing???

Reply:
The developer documentation doesn't call out the fact that some entities don't poll by default: https://developers.home-assistant.io/docs/core/entity#polling so I just assumed they all did. It's not mentioned on the binary sensor page either: https://developers.home-assistant.io/docs/core/entity/binary-sensor

Hence, I figured the callbacks wouldn't be called if I disabled polling so I just switched it to a property. 

It seems both of these may have been unnecessary.